src/Microsoft.Net.Http.Headers/HeaderUtilities.cs

```diff
@@ -208,6 +292,82 @@ public static bool TryParseInt64(string value, out long result)
             return long.TryParse(value, NumberStyles.None, NumberFormatInfo.InvariantInfo, out result);
         }
 
+        internal static unsafe bool TryParseInt32(StringSegment value, out int result)
```

Comment:
Not sure if we should use the same name as the older slower `TryParseInt32/64` since there are a few behaviour differences. We parse only positive numbers for example, and we don't check for overflows.